### PR TITLE
SSNSPA-220 retain filters

### PR DESCRIPTION
**Pull request by fabian202**

### CODE REVIEW

File: src/hooks/useSearch.js [hunk 1]
```diff
@@ -43,6 +45,7 @@ export const useSearch = () => {
   useEffect(() => {
     if (params) {
       executePrograms();
+      dispatch({ type: 'SET_PARAMS', payload: params });
```
Comment (fabian202) [inline src/hooks/useSearch.js:48]: Dispatch the action to maintain the params on the context

File: src/components/organisms/ScholarshipSearch/index.js [hunk 2]
```diff
@@ -32,6 +34,7 @@ const ScholarshipSearch = () => {
 
   const [organizations, setOrganizations] = useState();
   const [criterias, setCriterias] = useState();
+  const { state } = useProgramContext();
```
Comment (fabian202) [inline src/components/organisms/ScholarshipSearch/index.js:37]: Use the state from the hook context, if the action is dispatched at any other place, this state will take those changes, even if isn't a child of the component, so we can manage the state pretty much as redux does!

File: src/components/molecules/SearchPanel/index.js [hunk 3]
```diff
@@ -95,7 +131,7 @@ const SearchPanel = ({ organizations, criteria, onFilter, onClear }) => {
       handleFilter('globalSearch', globalSearch);
       setKeyWord(globalSearch);
     } else {
-      onFilter({});
+      // onFilter({});
```
Comment (cesar82) [inline src/components/molecules/SearchPanel/index.js:134]: remove it

File: src/components/molecules/SearchPanel/index.js [hunk 4]
```diff
@@ -51,8 +58,37 @@ const SearchPanel = ({ organizations, criteria, onFilter, onClear }) => {
   let typingTimeout,
     targetValue = false;
 
+  useEffect(() => {
+    //'citizenshipEligibilities', 'citizenshipStatuses'
```
Comment (normandanielmq) [inline src/components/molecules/SearchPanel/index.js:62]: Remove

File: src/components/molecules/SearchPanel/index.js [hunk 5]
```diff
@@ -396,9 +426,12 @@ const SearchPanel = ({ organizations, criteria, onFilter, onClear }) => {
           <Col>
             <StyledSelect
               options={filterDates}
-              defaultValue={filterDates[0]}
-              components={{ DropdownIndicator }}
               onChange={handleChange}
+              components={{ DropdownIndicator }}
```
Comment (normandanielmq) [inline src/components/molecules/SearchPanel/index.js:430]: Why do we need this? what is wrong with the default one?

File: src/components/molecules/SearchPanel/index.js [hunk 6]
```diff
@@ -602,7 +649,13 @@ const SearchPanel = ({ organizations, criteria, onFilter, onClear }) => {
...
               components={{ DropdownIndicator }}
               onChange={(e) => handleSelectChange(e, 'eligibilityCriteria')}
-              value={selects['eligibilityCriteria']}
+              value={
+                findDefaultSelect(
+                  defaultValue?.eligibilityCriteria ||
+                    selects['eligibilityCriteria'],
+                  eligibilityCriteria
+                ) || eligibilityCriteria[0]
+              }
```
Comment (normandanielmq) [inline src/components/molecules/SearchPanel/index.js:658]: This code is being duplicated in different locations above, consider building a function that returns this object
```
const buildDefaultFromSelects = (section, sectionName) => {
  return findDefaultSelet(
    defaultValue?.[sectionName] || selects[sectionName], 
    source
  ) || source[0];
}
```
And then: 
```
value={buildDefaultFromSelects(eligibilityCriteria, 'eligibilityCriteria')}
```
That function can be used in others selected above

File: src/components/molecules/SearchPanel/index.js [hunk 7]
```diff
@@ -635,7 +693,12 @@ const SearchPanel = ({ organizations, criteria, onFilter, onClear }) => {
               }))}
               onChange={(e) => handleMultiChange('races', e)}
               components={{ DropdownIndicator }}
-              value={multiples?.races}
+              value={
+                defaultValue?.races?.map((item) => ({
+                  value: item,
+                  label: item,
+                })) || multiples?.races
+              }
```
Comment (normandanielmq) [inline src/components/molecules/SearchPanel/index.js:701]: This one can be isolated in a function too

File: src/components/organisms/ScholarshipSearch/index.js [hunk 8]
```diff
@@ -109,7 +123,8 @@ const ScholarshipSearch = () => {
                 <ServerPagination
                   onChange={handleFilter}
                   {...dataPrograms?.meta}
-                  loading={loadingPrograms}
+                  $loading={loadingPrograms}
```
Comment (normandanielmq) [inline src/components/organisms/ScholarshipSearch/index.js:126]: `$loading?` Is this ok?

File: src/hooks/useProgramContext.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { useContext } from 'react';
+import { ProgramContext } from '../providers/programProvider';
+
+export const useProgramContext = () => {
+  const [state, dispatch] = useContext(ProgramContext);
+  return {
+    state,
+    dispatch,
+  };
+};
```
Comment (normandanielmq) [inline src/hooks/useProgramContext.js:10]: Feels like this function could live in the same file as the provider.  But renaming and moving the file into a `context` directory.  Take `searchContext` context in program-frontend as an example.

File: src/reducers/programReducer.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export const programReducer = (state, action) => {
+  switch (action.type) {
+    case 'SET_PARAMS':
+      return { ...state, params: action.payload };
+    case 'SET_PROGRAMS':
+      return { ...state, programs: action.payload };
+    default:
+      return state;
+  }
+};
```
Comment (normandanielmq) [inline src/reducers/programReducer.js:10]: This is cool, I have to read more about how this works. 💯

File: src/components/organisms/ScholarshipSearch/index.js (hunk 8 above)
Comment (fabian202) [reply src/components/organisms/ScholarshipSearch/index.js:126]: Norman this is a transient prop, is a prop that is not rendered on the DOM, so for example the component will not render loading=true

**Pull request merged**